State Assignment 

Questions on measure of central tendency

In [5]:
# 1.

import numpy as np
from scipy import stats as st

data=np.array([50,60,55,70])

mean = np.mean(data)
median = np.median(data)
mode = st.mode(data)

print(mean)
print(median)
print(mode)

58.75
57.5
ModeResult(mode=np.int64(50), count=np.int64(1))


Insights for the Retail Store:
The mean sales (58.75 units) indicate the average demand per week.
The median sales (57.5 units) represent the typical weekly sales performance.
Since there is no mode, there are no sales figures that occur more frequently.

In [6]:
# 2.

import numpy as np
from scipy import stats as st

data = np.array([15, 10, 20, 25, 15, 10, 30, 20, 15, 10, 10, 25, 15, 20, 20, 15, 10, 10, 20, 25])


print(np.mean(data))
print(np.median(data))
print(st.mode(data))

17.0
15.0
ModeResult(mode=np.int64(10), count=np.int64(6))


Insights for the Restaurant:
Since the mean (17 min) is higher than the median (15 min), there may be some longer wait times affecting the average.
The most frequent waiting time is 10 minutes, indicating that a significant number of customers get seated quickly.
The restaurant may need to analyze peak hours where wait times exceed 20+ minutes and optimize staffing accordingly.

In [7]:
# 3. 

import numpy as np
from scipy import stats as st

data = np.array([3, 2, 5, 4, 7, 2, 3, 3, 1, 6,
4, 2, 3, 5, 2, 4, 2, 1, 3, 5,
6, 3, 2, 1, 4, 2, 4, 5, 3, 2,
7, 2, 3, 4, 5, 1, 6, 2, 4, 3,
5, 3, 2, 4, 2, 6, 3, 2, 4, 5])

print(np.mean(data))
print(np.median(data))
print(st.mode(data))

3.44
3.0
ModeResult(mode=np.int64(2), count=np.int64(13))


Insights for the Car Rental Company:
Most customers rent cars for shorter durations (2-3 days), which suggests the need for flexible short-term rental plans.
The mean (3.44 days) being slightly higher than the median (3 days) indicates some longer rental durations influencing the average.
Optimizing fleet availability based on the frequent rental period (2-3 days) can improve efficiency and reduce downtime.
The company might consider discounts for longer rentals (5+ days) to balance fleet utilization and revenue.

Questions on measure of dispersion

In [ ]:
# 1. 

import numpy as np
import statistics as ss

data = np.array([120,110,130,115,125,105,135,115,125,140])

Large_val=140
Small_val=105
Range = Large_val - Small_val
print(Range)
print(ss.pvariance(data))
print(ss.stdev([120,110,130,115,125,105,135,115,125,140]))

35
111
11.105554165971787


Insights for the Manufacturing Company:
The range of 35 units indicates a noticeable fluctuation in machine output across the days, suggesting that the machine’s performance may vary.
The variance and standard deviation are also relatively high, indicating that the machine's performance is not highly consistent.
The company could consider improving quality control or performing routine maintenance to reduce output variability.
If production consistency is key, identifying and addressing the cause of fluctuations (e.g., machine calibration, material quality) might help stabilize output.

In [17]:
# 2. 
import statistics
import numpy as np

data = np.array([500, 700, 400, 600, 550, 750, 650, 500, 600, 550,800, 450, 700, 550, 600, 400, 650, 500, 750, 550,
700, 600, 500, 800, 550, 650, 400, 600, 750, 550])

Large_val=800
Small_val=400
Range = Large_val - Small_val
print(Range)
print(ss.pvariance(data))
print(ss.stdev([500, 700, 400, 600, 550, 750, 650, 500, 600, 550,800, 450, 700, 550, 600, 400, 650, 500, 750, 550,
700, 600, 500, 800, 550, 650, 400, 600, 750, 550]))

400
12725
114.73357443855863


Insights:

Range:
The range is the difference between the highest and lowest sales values. It shows the overall spread of sales, helping identify how extreme the fluctuations are over the given period.

Variance:
Variance measures how spread out the sales data is around the mean. A high variance indicates large fluctuations, while a low variance suggests consistent sales behavior.

Standard Deviation:
Standard deviation is the square root of variance, offering a more interpretable measure of spread in the same units as the data. It provides insights into how far individual sales are from the average, aiding in better forecasting.

In [18]:
# 3. 
import numpy as np
import statistics

data = np.array([3, 5, 2, 4, 6, 2, 3, 4, 2, 5,
7, 2, 3, 4, 2, 4, 2, 3, 5, 6,
3, 2, 1, 4, 2, 4, 5, 3, 2, 7,
2, 3, 4, 5, 1, 6, 2, 4, 3, 5,
3, 2, 4, 2, 6, 3, 2, 4, 5, 3])

Large_val=7
Small_val=1
Range = Large_val - Small_val
print(Range)
print(ss.pvariance(data))
print(ss.stdev([3, 5, 2, 4, 6, 2, 3, 4, 2, 5,
7, 2, 3, 4, 2, 4, 2, 3, 5, 6,
3, 2, 1, 4, 2, 4, 5, 3, 2, 7,
2, 3, 4, 5, 1, 6, 2, 4, 3, 5,
3, 2, 4, 2, 6, 3, 2, 4, 5, 3]))

6
2
1.528504671439458


These insights help assess the variability in shipment times, which can aid in improving logistics efficiency and managing customer expectations. Let me know if you need further analysis

In [ ]:
# 4. 

import numpy as np

data = np.array([120, 150, 110, 135, 125, 140, 130, 155, 115, 145, 135, 130])


print(np.mean(data))
Large_val=155
Small_val=110
Range = Large_val - Small_val
print(Range)

132.5
45


In [3]:
# 5. 


import numpy as np
import statistics as ss

data = np.array([8, 7, 9, 6, 7, 8, 9, 8, 7, 6,
8, 9, 7, 8, 7, 6, 8, 9, 6, 7,
8, 9, 7, 6, 7, 8, 9, 8, 7, 6,
9, 8, 7, 6, 8, 9, 7, 8, 7, 6,
9, 8, 7, 6, 7, 8, 9, 8, 7, 6])

print(np.mean(data))
print(ss.stdev([8, 7, 9, 6, 7, 8, 9, 8, 7, 6,
8, 9, 7, 8, 7, 6, 8, 9, 6, 7,
8, 9, 7, 6, 7, 8, 9, 8, 7, 6,
9, 8, 7, 6, 8, 9, 7, 8, 7, 6,
9, 8, 7, 6, 7, 8, 9, 8, 7, 6]))

7.5
1.0350983390135313


In [6]:
# 6.  

import numpy as np
import statistics as ss

data = np.array([10, 15, 12, 18, 20, 25, 8, 14, 16, 22,
9, 17, 11, 13, 19, 23, 21, 16, 24, 27,
13, 10, 18, 16, 12, 14, 19, 21, 11, 17,
15, 20, 26, 13, 12, 14, 22, 19, 16, 11,
25, 18, 16, 13, 21, 20, 15, 12, 19, 17,
14, 16, 23, 18, 15, 11, 19, 22, 17, 12,
16, 14, 18, 20, 25, 13, 11, 22, 19, 17,
15, 16, 13, 14, 18, 20, 19, 21, 17, 12,
15, 13, 16, 14, 22, 21, 19, 18, 16, 11,
17, 14, 12, 20, 23, 19, 15, 16, 13, 18])


print(np.mean(data))

Large_val=27
Small_val=8
Range = Large_val - Small_val

print(Range)
print(ss.stdev([10, 15, 12, 18, 20, 25, 8, 14, 16, 22,
9, 17, 11, 13, 19, 23, 21, 16, 24, 27,
13, 10, 18, 16, 12, 14, 19, 21, 11, 17,
15, 20, 26, 13, 12, 14, 22, 19, 16, 11,
25, 18, 16, 13, 21, 20, 15, 12, 19, 17,
14, 16, 23, 18, 15, 11, 19, 22, 17, 12,
16, 14, 18, 20, 25, 13, 11, 22, 19, 17,
15, 16, 13, 14, 18, 20, 19, 21, 17, 12,
15, 13, 16, 14, 22, 21, 19, 18, 16, 11,
17, 14, 12, 20, 23, 19, 15, 16, 13, 18]))

16.74
19
4.1429506881014655


In [8]:
# 7.  

import numpy as np
import statistics as ss

data = np.array([30, 32, 33, 28, 31, 30, 29, 30, 32, 31,
25, 27, 26, 23, 28, 24, 26, 25, 27, 28,
22, 23, 20, 25, 21, 24, 23, 22, 25, 24,
18, 17, 19, 20, 21, 18, 19, 17, 20, 19,
35, 36, 34, 35, 33, 34, 32, 33, 36, 34])


print(np.mean(data))

Large_val=36
Small_val=17
Range = Large_val - Small_val

print(Range)
print(ss.pvariance(data))

26.48
19
31


In [10]:
# 8. 

import numpy as np

data = np.array([28, 32, 35, 40, 42, 28, 33, 38, 30, 41,
37, 31, 34, 29, 36, 43, 39, 27, 35, 31,
39, 45, 29, 33, 37, 40, 36, 29, 31, 38,
35, 44, 32, 39, 36, 30, 33, 28, 41, 35,
31, 37, 42, 29, 34, 40, 31, 33, 38, 36,
39, 27, 35, 30, 43, 29, 32, 36, 31, 40,
38, 44, 37, 33, 35, 41, 30, 31, 39, 28,
45, 29, 33, 38, 34, 32, 35, 31, 40, 36,
39, 27, 35, 30, 43, 29, 32, 36, 31, 40,
38, 44, 37, 33, 35, 41, 30, 31, 39, 28])


print(np.mean(data))
print(np.median(data))

Large_val=27
Small_val=8
Range = Large_val - Small_val

print(Range)

34.99
35.0
19


In [12]:
# 9.  

import numpy as np
from scipy import stats as st

data = np.array([56, 40, 28, 73, 52, 61, 35, 40, 47, 65,
52, 44, 38, 60, 56, 40, 36, 49, 68, 57,
52, 63, 41, 48, 55, 42, 39, 58, 62, 49,
59, 45, 47, 51, 65, 41, 48, 55, 42, 39,
58, 62, 49, 59, 45, 47, 51, 65, 43, 58])


print(np.mean(data))
print(np.median(data))
print(st.iqr(data))

50.7
50.0
15.75


In [14]:
# 10.

from scipy import stats as st

data = np.array([30, 40, 20, 10, 45, 25, 30])

print(st.mode(data))

ModeResult(mode=np.int64(30), count=np.int64(2))


In [15]:
# 11.

from scipy import stats as st

data = np.array([4, 5, 3, 4, 4, 3, 2, 5, 4, 3,
5, 4, 2, 3, 4, 5, 3, 4, 5, 3,
4, 3, 2, 4, 5, 3, 4, 5, 4, 3,
3, 4, 5, 2, 3, 4, 4, 3, 5, 4,
3, 4, 5, 4, 2, 3, 4, 5, 3, 4,
5, 4, 3, 4, 5, 3, 4, 5, 4, 3,
3, 4, 5, 2, 3, 4, 4, 3, 5, 4,
3, 4, 5, 4, 2, 3, 4, 5, 3, 4,
5, 4, 3, 4, 5, 3, 4, 5, 4, 3,
3, 4, 5, 2, 3, 4, 4, 3, 5, 4])

print(st.mode(data))

ModeResult(mode=np.int64(4), count=np.int64(39))


In [17]:
# 12. 

import numpy as np
data = np.array([35, 28, 32, 45, 38, 29, 42, 30, 36, 41,
47, 31, 39, 43, 37, 30, 34, 39, 28, 33,
36, 40, 42, 29, 31, 45, 38, 33, 41, 35,
37,
34, 46, 30, 39, 43, 28, 32, 36, 29,
31, 37, 40, 42, 33, 39, 28, 35, 38, 43])

print(np.mean(data))

36.14


In [18]:
# 13. 

import numpy as np

data = np.array([ 125, 148, 137, 120, 135, 132, 145, 122, 130, 141,
118, 125, 132, 136, 128, 123, 132, 138, 126, 129,
136, 127, 130, 122, 125, 133, 140, 126, 133, 135,
130, 134, 141, 119, 125, 131, 136, 128, 124, 132,
136, 127, 130, 122, 125, 133, 140, 126, 133, 135,
130, 134, 141, 119, 125, 131, 136, 128, 124, 132,
136, 127, 130, 122, 125, 133, 140, 126, 133, 135,
130, 134, 141, 119, 125, 131, 136, 128, 124, 132,
136, 127, 130, 122, 125, 133, 140, 126, 133, 135,
130, 134, 141, 119, 125, 131, 136, 128, 124, 132])

print(np.median(data))

130.5


In [19]:
# 14. 

import numpy as np

data = np.array([  45, 35, 40, 38, 42, 37, 39, 43, 44, 41,
 32, 28, 30, 34, 33, 35, 31, 29, 36, 37,
 40, 39, 42, 41, 38, 43, 45, 44, 41, 37])

print(np.mean(data))

Large_val=45
Small_val=28
Range = Large_val - Small_val

print(Range)

37.96666666666667
17


Questions on Measure of Skewness and Kurtosis

In [ ]:
# 1. 

import pandas as pd

data = [-2.5, 1.3, -0.8, -1.9, 2.1, 0.5, -1.2, 1.8, -0.5, 2.3,
-0.7, 1.2, -1.5, -0.3, 2.6, 1.1, -1.7, 0.9, -1.4, 0.3,
1.9, -1.1, -0.4, 2.2, -0.9, 1.6, -0.6, -1.3, 2.4, 0.7,
-1.8, 1.5, -0.2, -2.1, 2.8, 0.8, -1.6, 1.4, -0.1, 2.5,
-1.0, 1.7, -0.9, -2.0, 2.7, 0.6, -1.4, 1.1, -0.3, 2.0]

df = pd.DataFrame(data)

print(df.skew())
print(df.kurt())

0    0.054546
dtype: float64


Interpretation:
Skewness (~0.05): The value is close to zero, indicating that the distribution of returns is approximately symmetric.
Kurtosis (-1.2951): The negative excess kurtosis suggests that the distribution is platykurtic, meaning it has lighter tails and is less peaked compared to a normal distribution. This implies fewer extreme returns (less risk of large losses or gains)

In [2]:
# 2. 

import pandas as pd

data = [2.5, 4.8, 3.2, 2.1, 4.5, 2.9, 2.3, 3.1, 4.2, 3.9,
2.8, 4.1, 2.6, 2.4, 4.7, 3.3, 2.7, 3.0, 4.3, 3.7,
2.2, 3.6, 4.0, 2.7, 3.8, 3.5, 3.2, 4.4, 2.0, 3.4,
3.1, 2.9, 4.6, 3.3, 2.5, 4.9, 2.8, 3.0, 4.2, 3.9,
2.8, 4.1, 2.6, 2.4, 4.7, 3.3, 2.7, 3.0, 4.3, 3.7,
2.2, 3.6, 4.0, 2.7, 3.8, 3.5, 3.2, 4.4,
2.0, 3.4,
3.1, 2.9, 4.6, 3.3, 2.5, 4.9, 2.8, 3.0, 4.2, 3.9,
2.8, 4.1, 2.6, 2.4, 4.7, 3.3, 2.7, 3.0, 4.3, 3.7,
2.2, 3.6, 4.0, 2.7, 3.8, 3.5, 3.2, 4.4, 2.0, 3.4,
3.1, 2.9, 4.6, 3.3, 2.5, 4.9]

df = pd.DataFrame(data)

print(df.skew())
print(df.kurt())

0    0.224025
dtype: float64
0   -0.931209
dtype: float64


Interpretation:
Skewness (~0.224025): The value is close to zero, indicating that the distribution of returns is approximately symmetric.
Kurtosis (-0.931209): The negative excess kurtosis suggests that the distribution is platykurtic, meaning it has lighter tails and is less peaked compared to a normal distribution. This implies fewer extreme returns (less risk of large losses or gains)

In [3]:
# 3. 

import pandas as pd

data = [4, 5, 3, 4, 4, 3, 2, 5, 4, 3,
5, 4, 2, 3, 4, 5, 3, 4, 5, 3,
4, 3, 2, 4, 5, 3, 4, 5, 4, 3,
3, 4, 5, 2, 3, 4, 4, 3, 5, 4,
3, 4, 5, 4, 2, 3, 4, 5, 3, 4,
5, 4, 3, 4, 5, 3, 4, 5, 4, 3,
3, 4, 5, 2, 3, 4, 4, 3, 5, 4,
3, 4, 5, 4, 2, 3, 4, 5, 3, 4,
5, 4, 3, 4, 5, 3, 4, 5, 4, 3,
3, 4, 5, 2, 3, 4, 4, 3, 5, 4]

df = pd.DataFrame(data)

print(df.skew())
print(df.kurt())

0   -0.21091
dtype: float64
0   -0.745256
dtype: float64


Interpretation:
Skewness (-0.21091): The value is close to zero, indicating that the distribution of returns is approximately symmetric.
Kurtosis (-0.745256): The negative excess kurtosis suggests that the distribution is platykurtic, meaning it has lighter tails and is less peaked compared to a normal distribution. This implies fewer extreme returns (less risk of large losses or gains)

In [4]:
# 4. 

import pandas as pd

data = [280, 350, 310, 270, 390, 320, 290, 340, 310, 380,
270, 350, 300, 330, 370, 310, 280, 320, 350, 290,
270, 350, 300, 330, 370, 310, 280, 320, 350, 290,
270, 350, 300, 330, 370, 310, 280, 320, 350, 290,
270, 350, 300, 330, 370, 310, 280, 320, 350, 290,
270, 350, 300, 330, 370, 310, 280, 320, 350, 290,
270, 350, 300, 330, 370, 310, 280, 320, 350, 290,
270, 350, 300, 330, 370, 310, 280, 320, 350, 290,
270, 350, 300, 330, 370, 310, 280, 320, 350, 290,
270, 350, 300, 330, 370, 310, 280, 320, 350, 290]

df = pd.DataFrame(data)

print(df.skew())
print(df.kurt())

0    0.209219
dtype: float64
0   -1.037424
dtype: float64


Interpretation:
Skewness (~0.209219): The value is close to zero, indicating that the distribution of returns is approximately symmetric.
Kurtosis (-1.037424): The negative excess kurtosis suggests that the distribution is platykurtic, meaning it has lighter tails and is less peaked compared to a normal distribution. This implies fewer extreme returns (less risk of large losses or gains)

In [5]:
# 5. 

import pandas as pd

data = [12, 18, 15, 22, 20, 14, 16, 21, 19, 17,
22, 19, 13, 16, 21, 22, 17, 19, 22, 18,
14, 20, 19, 17, 22, 18, 15, 21, 20, 16,
12, 18, 15, 22, 20, 14, 16, 21, 19, 17,
22, 19, 13, 16, 21, 22, 17, 19, 22, 18,
14, 20, 19, 17, 22, 18, 15, 21, 20, 16,
12, 18, 15, 22, 20, 14, 16, 21, 19, 17,
22, 19, 13, 16, 21, 22, 17, 19, 22, 18,
14, 20, 19, 17, 22, 18, 15, 21, 20, 16,
12, 18, 15, 22, 20, 14, 16, 21, 19, 17]

df = pd.DataFrame(data)

print(df.skew())
print(df.kurt())

0   -0.335013
dtype: float64
0   -0.881011
dtype: float64


Interpretation:
Skewness (-0.335013): The value is close to zero, indicating that the distribution of returns is approximately symmetric.
Kurtosis (-0.881011): The negative excess kurtosis suggests that the distribution is platykurtic, meaning it has lighter tails and is less peaked compared to a normal distribution. This implies fewer extreme returns (less risk of large losses or gains)

Questions on Percentile and Quartiles

In [8]:
import numpy as np 

data = [40, 45, 50, 55, 60, 62, 65, 68, 70, 72,
75, 78, 80, 82, 85, 88, 90, 92, 95, 100,
105, 110, 115, 120, 125, 130, 135, 140, 145, 150,
155, 160, 165, 170, 175, 180, 185, 190, 195, 200,
205, 210, 215, 220, 225, 230, 235, 240, 245, 250,
255, 260, 265, 270, 275, 280, 285, 290, 295, 300,
305, 310, 315, 320, 325, 330, 335, 340, 345, 350,
355, 360, 365, 370, 375, 380, 385, 390, 395, 400,
405, 410, 415, 420, 425, 430, 435, 440, 445, 450,
455, 460, 465, 470, 475, 480, 485, 490, 495, 500]

a = np.quantile(data,[0, 0.10, 0.25, 0.5, 0.75, 0.90, 1])

print(a)

[ 40.    74.7  128.75 252.5  376.25 450.5  500.  ]


In [9]:
import numpy as np 

data = [55, 60, 62, 65, 68, 70, 72, 75, 78, 80,
82, 85, 88, 90, 92, 95, 100, 105, 110, 115,
120, 125, 130, 135, 140, 145, 150, 155, 160, 165,
170, 175, 180, 185, 190, 195, 200, 205, 210, 215,
220, 225, 230, 235, 240, 245, 250, 255, 260, 265,
270, 275, 280, 285, 290, 295, 300, 305, 310, 315,
320, 325, 330, 335, 340, 345, 350, 355, 360, 365,
370, 375,
380, 385, 390, 395, 400, 405, 410, 415,
420, 425, 430, 435, 440, 445, 450, 455, 460, 465,
470, 475, 480, 485, 490, 495, 500, 505, 510, 515]

a = np.quantile(data,[0, 0.15, 0.25, 0.5, 0.75, 0.85, 1])

print(a)

[ 55.    94.55 143.75 267.5  391.25 440.75 515.  ]


Questions on Correlation and Covariance

In [11]:
import numpy as np

ae = np.array([10, 12, 15, 18, 20, 22, 25, 28, 30, 32, 35, 38])
sr = np.array([50, 55, 60, 65, 70, 75, 80, 85, 90, 95, 100, 105])

cf = np.corrcoef(ae, sr)[0, 1]

print(cf)


0.9992103100366481


In [ ]:
import numpy as np

a = np.array([ 45, 47, 48, 50, 52, 53, 55, 56, 58, 60, 62, 64, 65, 67, 69, 70, 72, 74, 76, 77])
b = np.array([ 52, 54, 55, 57, 59, 60, 61, 62, 64, 66, 67, 69, 71, 73, 74, 76, 78, 80, 82, 83])

cf = np.corrcoef(a, b)[0, 1]

print(cf)

0.9985957269963789


Questions on discrete and continuous random variable

Discrete Random Variable

1.Rolling exactly five 3's in 100 rolls of a fair six-sided die:
≈ 0.00029 (0.029%) – Very unlikely.


2.Drawing exactly two hearts in a 5-card hand from a 52-card deck:
≈ 0.274 (27.4%) – Fairly likely.


3.Getting at least 8 correct answers by randomly guessing in a 10-question multiple-choice test:
≈ 0.00042 (0.042%) – Extremely rare.


4.Drawing 3 blue balls from a bag containing 20 blue, 30 red, and 10 green balls:
≈ 0.0333 (3.33%) – Unlikely, but possible.


5.Scoring exactly 3 goals in 10 shots with a 30% probability per shot:
≈ 0.267 (26.7%) – Relatively common.

Continuous Random Variable

1.Probability that a randomly selected student is taller than 180 cm:
≈ 0.067 (6.68%) – Uncommon, but possible.


2.Probability that a customer waits less than 3 minutes in a coffee shop:
≈ 0.451 (45.1%) – Almost a 50/50 chance.


3.Probability that a light bulb lasts between 900 and 1100 hours:
≈ 0.683 (68.3%) – Quite likely.


4.Probability that an apple weighs between 150 and 170 grams in a uniform distribution:
≈ 0.20 (20%) – Fairly low.


5.Probability that a task is completed in less than 15 minutes:
≈ 0.528 (52.8%) – Slightly more than half the time.

Questions on Hypothesis Testings

In [1]:
import numpy as np
from scipy import stats

new = np.array([85, 90, 78, 92, 88, 76, 95, 89, 84, 91])
old = np.array([80, 85, 74, 86, 82, 70, 88, 83, 79, 84])

t_stat, p_value = stats.ttest_ind(new, old)

print(f"T-statistic: {t_stat:.4f}")
print(f"P-value: {p_value:.4f}")

alpha = 0.05 
if p_value < alpha:
    print("Reject the null hypothesis: There is a significant difference between the groups.")
else:
    print("Fail to reject the null hypothesis: No significant difference found between the groups.")

T-statistic: 2.1850
P-value: 0.0424
Reject the null hypothesis: There is a significant difference between the groups.


In [3]:
import scipy.stats as stats

x_bar = 510   
mu = 500      
s = 20        
n = 25       
t_stat = (x_bar - mu) / (s / (n ** 0.5))

p_value = 2 * (1 - stats.t.cdf(abs(t_stat), df=n-1))

print(f"t-statistic: {t_stat:.4f}")
print(f"p-value: {p_value:.4f}")

alpha = 0.05
if p_value < alpha:
    print("Reject the null hypothesis (H0): There is significant evidence that the population mean is different from 500 grams.")
else:
    print("Fail to reject the null hypothesis (H0): There is not enough evidence to conclude a difference in mean weight.")


t-statistic: 2.5000
p-value: 0.0197
Reject the null hypothesis (H0): There is significant evidence that the population mean is different from 500 grams.
